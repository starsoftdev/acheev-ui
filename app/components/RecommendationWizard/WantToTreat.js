// @flow
import * as React from 'react';
import cx from 'classnames';

import Button from 'components/Button';
import Icon from 'components/Icon';
import PainIcon from 'images/sprite/pain.svg';
import AnxietyIcon from 'images/sprite/anxiety.svg';
import InsomniaIcon from 'images/sprite/insomnia.svg';
import AppetiteIcon from 'images/sprite/appetite.svg';
import DepressionIcon from 'images/sprite/depression.svg';
import ArrowIcon from './arrow.png';

import Block from './Block';

const options = [
  {
    icon: PainIcon,
    label: 'Pain',
    field: 'pain',
  },
  {
    icon: AnxietyIcon,
    label: 'Anxiety',
    field: 'anxiety',
  },
  {
    icon: InsomniaIcon,
    label: 'Insomnia',
    field: 'insomnia',
  },
  {
    icon: DepressionIcon,
    label: 'Depression',
    field: 'depression',
  },
  {
    icon: AppetiteIcon,
    label: 'Lack of appetite',
    field: 'appetite',
  },
];

type Props = {
  onSelect: (value: Array<string>) => void,
  // eslint-disable-next-line  react/no-unused-prop-types
  value: Array<string>,
};

type State = {
  pain: boolean,
  insomnia: boolean,
  depression: boolean,
  appetite: boolean,
  anxiety: boolean,
  showMessage: boolean,
};

export default class WantToTreat extends React.Component<Props, State> {
  state = {
    pain: false,
    insomnia: false,
    depression: false,
    appetite: false,
    anxiety: false,
    showMessage: false,
  };

  handleClick = (e: SyntheticEvent<HTMLInputElement>) => {
    const { name } = e.currentTarget.dataset;
    this.setState(state => ({ [name]: !state[name], showMessage: false }));
  };

  handleFindClick = () => {
    const fields = { ...this.state };
    delete fields.showMessage;

    if (Object.values(fields).every(status => !status)) {
      this.setState({ showMessage: true });
    } else {
      const values = Object.keys(fields).filter(field => fields[field]);
      this.props.onSelect(values);
    }
  };

  render() {
    return (
      <section className="recommendationWizard__treatContainer">
        <div className="container">
          <div className="row">
            <h1 className="recommendationWizard__title small-12 column">
              What do you want to treat?
            </h1>
            <div className="small-12 column">
              <div className="row">
                {options.map((item, index) => (
                  <div
                    key={item.field}
                    className={cx('small-12 large-2 column', {
                      'large-offset-1': index === 0,
                    })}
                  >
                    <Block
                      label={item.label}
                      onClick={this.handleClick}
                      data-name={item.field}
                      selected={this.state[item.field]}
                    >
                      <Icon
                        glyph={item.icon}
                        width={100}
                        height={35}
                        className="recommendationWizard__icon"
                      />
                    </Block>
                  </div>
                ))}
              </div>
            </div>
          </div>
        </div>

        <div className="recommendationWizard--center recommendationWizard__message">
          {this.state.showMessage && 'You must select at least one symptom'}
        </div>

        <div className="recommendationWizard--center recommendationWizard__firstButton__wrapper">
          <Button
            className="recommendationWizard__firstButton coral"
            onClick={this.handleFindClick}
          >
            Find Strains
            <img
              src={ArrowIcon}
              alt="next_icon"
              className="recommendationWizard__buttonIcon"
            />
          </Button>
        </div>
      </section>
    );
  }
}
