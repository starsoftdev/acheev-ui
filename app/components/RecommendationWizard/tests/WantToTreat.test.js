import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import WantToTreat from 'components/RecommendationWizard/WantToTreat';

describe('<WantToTreat />', () => {
  it('should match snapshot', () => {
    const component = create(<WantToTreat onSelect={() => {}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should show error', () => {
    const handleSelect = jest.fn();
    const component = shallow(<WantToTreat onSelect={handleSelect} />);
    component.find('Button').simulate('click');
    expect(component.state().showMessage).toEqual(true);
    expect(handleSelect).toHaveBeenCalledTimes(0);
  });

  it('Should call onSelect', () => {
    const handleSelect = jest.fn();
    const component = shallow(<WantToTreat onSelect={handleSelect} />);
    component
      .find('Block')
      .first()
      .simulate('click', { currentTarget: { dataset: { name: 'pain' } } });
    component.find('Button').simulate('click');
    expect(component.state().showMessage).toEqual(false);
    expect(handleSelect).toHaveBeenCalledWith(['pain']);
    expect(handleSelect).toHaveBeenCalledTimes(1);
  });
});
