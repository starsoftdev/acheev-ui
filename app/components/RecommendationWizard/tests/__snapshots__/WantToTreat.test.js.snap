// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WantToTreat /> should match snapshot 1`] = `
<section
  className="recommendationWizard__treatContainer"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <h1
        className="recommendationWizard__title small-12 column"
      >
        What do you want to treat?
      </h1>
      <div
        className="small-12 column"
      >
        <div
          className="row"
        >
          <div
            className="small-12 large-2 column large-offset-1"
          >
            <div
              className="recommendationWizard__block"
              data-name="pain"
              onClick={[Function]}
            >
              <svg
                className="recommendationWizard__icon"
                height={35}
                width={100}
              >
                <use
                  xlinkHref="#undefined"
                />
              </svg>
              <p
                className="recommendationWizard__label"
              >
                Pain
              </p>
            </div>
          </div>
          <div
            className="small-12 large-2 column"
          >
            <div
              className="recommendationWizard__block"
              data-name="anxiety"
              onClick={[Function]}
            >
              <svg
                className="recommendationWizard__icon"
                height={35}
                width={100}
              >
                <use
                  xlinkHref="#undefined"
                />
              </svg>
              <p
                className="recommendationWizard__label"
              >
                Anxiety
              </p>
            </div>
          </div>
          <div
            className="small-12 large-2 column"
          >
            <div
              className="recommendationWizard__block"
              data-name="insomnia"
              onClick={[Function]}
            >
              <svg
                className="recommendationWizard__icon"
                height={35}
                width={100}
              >
                <use
                  xlinkHref="#undefined"
                />
              </svg>
              <p
                className="recommendationWizard__label"
              >
                Insomnia
              </p>
            </div>
          </div>
          <div
            className="small-12 large-2 column"
          >
            <div
              className="recommendationWizard__block"
              data-name="depression"
              onClick={[Function]}
            >
              <svg
                className="recommendationWizard__icon"
                height={35}
                width={100}
              >
                <use
                  xlinkHref="#undefined"
                />
              </svg>
              <p
                className="recommendationWizard__label"
              >
                Depression
              </p>
            </div>
          </div>
          <div
            className="small-12 large-2 column"
          >
            <div
              className="recommendationWizard__block"
              data-name="appetite"
              onClick={[Function]}
            >
              <svg
                className="recommendationWizard__icon"
                height={35}
                width={100}
              >
                <use
                  xlinkHref="#undefined"
                />
              </svg>
              <p
                className="recommendationWizard__label"
              >
                Lack of appetite
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="recommendationWizard--center recommendationWizard__message"
  />
  <div
    className="recommendationWizard--center recommendationWizard__firstButton__wrapper"
  >
    <a
      className="button recommendationWizard__firstButton coral"
      onClick={[Function]}
    >
      <div
        className="row align-middle align-center"
      >
        <div
          className="shrink column"
        >
          Find Strains
          <img
            alt="next_icon"
            className="recommendationWizard__buttonIcon"
            src="IMAGE_MOCK"
          />
        </div>
      </div>
    </a>
  </div>
</section>
`;
